lib/key/utils.js

```diff
@@ -101,3 +104,67 @@ export async function cloneKey(inputKey) {
     const [key] = await getKeys(inputKey.toPacketlist().write());
     return key;
 }
+
+/**
+ * Generate ECDHE key and secret from public key
+ *
+ * @param {Array<Options>}                              Public key Q, fingerprint and curve (name or OID)
+ * @returns {Promise<{V: Uint8Array, Z: Uint8Array}>}   Returns public part of ephemeral key and generated ephemeral secret
+ * @async
+ */
+export async function genPublicEphemeralKey(options) {
+    const curveObj = new openpgp.crypto.publicKey.elliptic.Curve(options.Curve);
+    const oid = new openpgp.OID(curveObj.oid);
+    const { V, S } = await openpgp.crypto.publicKey.elliptic.ecdh.genPublicEphemeralKey(curveObj, options.Q);
```

Comment:
No need to wrap it with try/catch ? Is it expected ?

Reply:
It is try/catched upstream in the server